aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -350,10 +352,17 @@ public void confirmSignUp(
     @Override
     public void resendSignUpCode(
             @NonNull String username,
+            @NonNull AuthResendSignUpCodeOptions options,
```

Comment:
Isn't it possible to introduce another overloaded function instead of inserting new NonNull param in a public function?  If someone was calling this function, then they would also need to change as a result of this change.  For public SDK's it's best to minimize breaking changes.

Reply:
This PR does introduce a new overloaded function. The diff here is a possibly a little misleading since it shows `resendSignUpCode` being modified. `resendSignUpCode` definied on like 381 below has the same method signature as before, and it calls this newly added method with a default option. This will not be a breaking change.